packages/workflow/src/worker-interface.ts

```diff
@@ -79,7 +79,21 @@ export function initRuntime(options: WorkflowCreateOptionsInternal): void {
     const workflowFn = mod[activator.info.workflowType];
     const defaultWorkflowFn = mod['default'];
 
-    if (typeof workflowFn === 'function') {
+    if (isWorkflowFunctionWithOptions(workflowFn)) {
```

Comment:
We're loosing some important type checking here, and making the "default export with options" win over "exact named function without options", which seems undesirable.

Keep the `(typeof workflowFn === 'function')` and `(typeof defaultWorkflowFn === 'function')` at the top level, and move the check for options inside those top level branches.

Also, in the `else` branch of `workflowFn.options`, make sure that it is indeed a function before assigning it to `workflowDefinitionOptionsGetter`.